sqlglot/generator.py

```diff
@@ -318,8 +318,21 @@ def insert_sql(self, expression):
         kind = "OVERWRITE" if expression.args.get("overwrite") else "INTO"
         this = self.sql(expression, "this")
         exists = " IF EXISTS " if expression.args.get("exists") else " "
+        partitions = expression.args.get("partition")
+        if partitions:
+            if isinstance(partitions[0], tuple):
+                partition_fields = ", ".join(
```

Comment:
i think i have a helper method, called csv, also can't you just do the check?

csv(f"{x}={y}" if isinstance(partition, tupel) else self.sql(partition) for partition in partitions)

Reply:
Re-implemented this so we don't have to check instancetype